tool/tctl/common/resource_command.go

```diff
@@ -276,7 +278,7 @@ func (rc *ResourceCommand) Create(ctx context.Context, client auth.ClientI) (err
 		if !found {
 			// if we're trying to create an OIDC/SAML connector with the OSS version of tctl, return a specific error
 			if raw.Kind == "oidc" || raw.Kind == "saml" {
```

Comment:
This branch is never taken now that `rc.CreateHandlers` has entries for SAML and OIDC. We emit a less useful error message now, and given a recent change to the SAML error message but not yet to OIDC, the SAML errors are also yuck:

```
$ tctl create -f r/connector-okta-dev.yaml
ERROR: rpc error: code = Unknown desc = SAML: this feature requires Teleport Enterprise

$ tctl create -f r/connector-oidc-auth0.yaml
ERROR: OIDC is only available in enterprise subscriptions
```

I'm trying to figure out where all the wrapped errors get converted to a GRPC status and why we get different error messages for each of these. Underneath they both wrap `trace,AccessDenied`, but OIDC returns the `trace.AccessDenied` error directly whereas SAML uses `trace.Wrap` on it.

If we want to still output this specific error message, we're going to have to either put it in the server or identify AccessDenied errors when trying to create these resources.

Reply:
Finally got back to this today.

Unfortunately, the `trace` package has a number of nuances here that prevent us from being able to propagate wrapped errors over the gRPC interface.

I opted to preserve the clean user-facing error message (without the confusing RPC error stuff in there), even though it means we can't wrap `ErrRequiresEnterprise` for now.